### PR TITLE
sana accepts multi-format of graphs as its input

### PR DESCRIPTION
**Pull request by xiaoyao007**

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -1,45 +1,46 @@
...
-    }else{
-        scheduleOnly = false;
-    }
-    cout << "Seed: " << getRandomSeed() << endl;
-    Mode* mode = selectMode(args);
-    mode->run(args);
-    delete mode;
-    return 0;
-}
+#include "arguments/ArgumentParser.hpp"
```
Comment (khosravipasha) [inline src/main.cpp:1]: Somehow git thinks all the lines are changed, I suspect its a spacing difference. 

Please make sure to fix this that the diff only includes the lines you changes.

File: src/convert.cpp [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+int find_type(string input)
+{
+	regex graphml(".*(\\.xml)");
+	regex gml(".*(\\.gml)");
+	regex leda(".*(\\.leda)");
+	regex csv(".*(\\.csv)");
+	regex lgf(".*(\\.lgf)");
+	regex edgelist(".*(\\.el)");
+	if(regex_match(input,lgf))
+	return 6;
```
Comment (khosravipasha) [inline src/convert.cpp:16]: Nit pick: Indentation is off here.

File: src/convert.cpp [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+
+string  convert(string filename)
+ {
+
+	string result = " tmpel";
+    switch(find_type(filename))
+{
+	
+	case 0: 
+	{cout << "Wrong" << endl;
```
Comment (khosravipasha) [inline src/convert.cpp:315]: Maybe add a explanation of what went wrong.

File: src/convert.cpp [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+}
+
+
+
+
+void convert_csv( string filename){
+	ifstream ifs;
+	ifs.open(filename, ifstream::in);
+	ofstream ofs;
+	ofs.open(" tmpel");
```
Comment (khosravipasha) [inline src/convert.cpp:228]: In `ofs.open(" tmpel");`
Extra space in the name, also more importantly hard coding the file name is not good idea, if we run multiple sana's they would try to write to same file and fail. 

Should probably be a combination of a random number + the original filename.

File: src/convert.cpp [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+	case 5: 
+	{	convert_leda(filename);
+		break;}
+	case 6: 
+	{	convert_lgf(filename);
+		break;}
+
+}
+
+    return result;
```
Comment (khosravipasha) [inline src/convert.cpp:339]: `result` is not set anywhere other that its initialization

**Review by khosravipasha — CHANGES_REQUESTED** [review]
Few requested changes in line. See each individual comment.

Also on a high level, I think this should be done in a different way for performace reasons. Currently if the graph is not a ".el" files we would read the whole graph and convert it to an edgelist and then write the edgelist to file and then finally give the edgelist to sana, for small graph its okay but for bigger ones it would be really slow as we would need to read the graph twice.

File: src/convert.cpp (hunk 4 above)
Comment (beneaththeglamour) [reply src/convert.cpp:228]: Yes I suggest using getpid()

**Comment by waynebhayes** [comment]
Yeah many things need changing; creating a temporary .el file is OK but it has to have a unique name.  I'm rejecting this pull request, please fix problems above.

**Pull request closed without merging**